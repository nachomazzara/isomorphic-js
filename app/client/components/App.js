import React from 'react'
import {
  BrowserRouter as Router,
  Redirect,
  Route,
  Switch
} from 'react-router-dom'

import TwitList from './TwitList'
import Twit from './Twit'

const appStyle = {
  width: '100%',
  maxWidth: '1200px',
  margin: 'auto',
  fontFamily: 'Roboto',
}

export default function App (props) {

  const { twits } = props
  return (
    <div style={appStyle}>
      <Switch>
        <Route path="/" render={(location) => (<TwitList twits={twits} location={location} />)} />
      </Switch>
    </div>
  )
}
