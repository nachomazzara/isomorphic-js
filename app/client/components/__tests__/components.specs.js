import React from 'react';
import { shallow, mount, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { StaticRouter, Redirect } from 'react-router-dom'

import App from '../App'

configure({ adapter: new Adapter() })

const mockTwitList = [
  {
    'id_str': 1234,
    'user': {}
  },
  {
    'id_str': 5678,
    'user': {}
  }
]

const mockTwit = [
  {
    'id_str': 1234,
    'user': {}
  }
]

describe('App', () => {
  it('should render Empty List of twits for route /', () => {
    const component = mount(
      <StaticRouter context={{}} location={'/'}>
        <App twits={[]}/>
      </StaticRouter>
    )
    expect(component).toMatchSnapshot()
  })

  it('should render List of twits for route /', () => {
    const component = mount(
      <StaticRouter context={{}} location={'/'}>
        <App twits={mockTwitList}/>
      </StaticRouter>
    )
    expect(component).toMatchSnapshot()
  })

  it('should render Twit specific for route /twit/ID', () => {
    const component = mount(
      <StaticRouter context={{}} location={'/twit/1234'}>
        <App twits={mockTwit} />
      </StaticRouter>
    )
    expect(component).toMatchSnapshot()
  })
})
