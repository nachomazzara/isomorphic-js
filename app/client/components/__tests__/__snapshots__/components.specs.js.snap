// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render Empty List of twits for route / 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App
      twits={Array []}
    >
      <div
        style={
          Object {
            "fontFamily": "Roboto",
            "margin": "auto",
            "maxWidth": "1200px",
            "width": "100%",
          }
        }
      >
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
              }
            }
            path="/"
            render={[Function]}
          >
            <TwitList
              location={
                Object {
                  "history": Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  },
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                  },
                  "match": Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  },
                  "staticContext": Object {},
                }
              }
              twits={Array []}
            >
              <div
                style={
                  Object {
                    "marginTop": "30px",
                  }
                }
              >
                <input
                  onChange={[Function]}
                  placeholder="Enter Search Criteria"
                  type="text"
                  value=""
                />
                <button
                  onClick={[Function]}
                >
                  Search
                </button>
                Enter  Search criteria in order to retrieve Twits
              </div>
            </TwitList>
          </Route>
        </Switch>
      </div>
    </App>
  </Router>
</StaticRouter>
`;

exports[`App should render List of twits for route / 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App
      twits={
        Array [
          Object {
            "id_str": 1234,
            "user": Object {},
          },
          Object {
            "id_str": 5678,
            "user": Object {},
          },
        ]
      }
    >
      <div
        style={
          Object {
            "fontFamily": "Roboto",
            "margin": "auto",
            "maxWidth": "1200px",
            "width": "100%",
          }
        }
      >
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
              }
            }
            path="/"
            render={[Function]}
          >
            <TwitList
              location={
                Object {
                  "history": Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  },
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                  },
                  "match": Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  },
                  "staticContext": Object {},
                }
              }
              twits={
                Array [
                  Object {
                    "id_str": 1234,
                    "user": Object {},
                  },
                  Object {
                    "id_str": 5678,
                    "user": Object {},
                  },
                ]
              }
            >
              <div
                style={
                  Object {
                    "marginTop": "30px",
                  }
                }
              >
                <input
                  onChange={[Function]}
                  placeholder="Enter Search Criteria"
                  type="text"
                  value=""
                />
                <button
                  onClick={[Function]}
                >
                  Search
                </button>
                <ul>
                  <li
                    key="1234"
                    style={
                      Object {
                        "listStyle": "none",
                      }
                    }
                  >
                     
                    <Twit
                      handleDelete={[Function]}
                      twit={
                        Object {
                          "id_str": 1234,
                          "user": Object {},
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "border": "3px solid #b9e0ff",
                            "borderRadius": "10px",
                            "display": "flex",
                            "marginBottom": "30px",
                            "padding": "15px",
                          }
                        }
                      >
                        <div>
                          <img
                            style={
                              Object {
                                "borderRadius": "50%",
                                "height": "auto",
                                "width": "50px",
                              }
                            }
                          />
                        </div>
                        <div
                          style={
                            Object {
                              "marginLeft": " 20px",
                            }
                          }
                        >
                          <span>
                            <strong />
                             @undefined
                          </span>
                          <p
                            style={
                              Object {
                                "margin": "3px 0",
                              }
                            }
                          />
                          <a
                            href="/twit/1234"
                            style={
                              Object {
                                "color": "blue",
                                "fontSize": "16px",
                                "margin": "10px",
                              }
                            }
                          >
                             
                            View
                             
                          </a>
                          <button
                            onClick={[Function]}
                            style={
                              Object {
                                "background": "transparent",
                                "border": "none",
                                "fontSize": "16px",
                                "margin": "10px 5px",
                              }
                            }
                          >
                            Delete
                          </button>
                        </div>
                      </div>
                    </Twit>
                     
                  </li>
                  <li
                    key="5678"
                    style={
                      Object {
                        "listStyle": "none",
                      }
                    }
                  >
                     
                    <Twit
                      handleDelete={[Function]}
                      twit={
                        Object {
                          "id_str": 5678,
                          "user": Object {},
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "border": "3px solid #b9e0ff",
                            "borderRadius": "10px",
                            "display": "flex",
                            "marginBottom": "30px",
                            "padding": "15px",
                          }
                        }
                      >
                        <div>
                          <img
                            style={
                              Object {
                                "borderRadius": "50%",
                                "height": "auto",
                                "width": "50px",
                              }
                            }
                          />
                        </div>
                        <div
                          style={
                            Object {
                              "marginLeft": " 20px",
                            }
                          }
                        >
                          <span>
                            <strong />
                             @undefined
                          </span>
                          <p
                            style={
                              Object {
                                "margin": "3px 0",
                              }
                            }
                          />
                          <a
                            href="/twit/5678"
                            style={
                              Object {
                                "color": "blue",
                                "fontSize": "16px",
                                "margin": "10px",
                              }
                            }
                          >
                             
                            View
                             
                          </a>
                          <button
                            onClick={[Function]}
                            style={
                              Object {
                                "background": "transparent",
                                "border": "none",
                                "fontSize": "16px",
                                "margin": "10px 5px",
                              }
                            }
                          >
                            Delete
                          </button>
                        </div>
                      </div>
                    </Twit>
                     
                  </li>
                </ul>
              </div>
            </TwitList>
          </Route>
        </Switch>
      </div>
    </App>
  </Router>
</StaticRouter>
`;

exports[`App should render Twit specific for route /twit/ID 1`] = `
<StaticRouter
  basename=""
  context={Object {}}
  location="/twit/1234"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/twit/1234",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <App
      twits={
        Array [
          Object {
            "id_str": 1234,
            "user": Object {},
          },
        ]
      }
    >
      <div
        style={
          Object {
            "fontFamily": "Roboto",
            "margin": "auto",
            "maxWidth": "1200px",
            "width": "100%",
          }
        }
      >
        <Switch>
          <Route
            computedMatch={
              Object {
                "isExact": false,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            location={
              Object {
                "hash": "",
                "pathname": "/twit/1234",
                "search": "",
              }
            }
            path="/"
            render={[Function]}
          >
            <TwitList
              location={
                Object {
                  "history": Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/twit/1234",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  },
                  "location": Object {
                    "hash": "",
                    "pathname": "/twit/1234",
                    "search": "",
                  },
                  "match": Object {
                    "isExact": false,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  },
                  "staticContext": Object {},
                }
              }
              twits={
                Array [
                  Object {
                    "id_str": 1234,
                    "user": Object {},
                  },
                ]
              }
            >
              <div
                style={
                  Object {
                    "marginTop": "30px",
                  }
                }
              >
                <input
                  onChange={[Function]}
                  placeholder="Enter Search Criteria"
                  type="text"
                  value=""
                />
                <button
                  onClick={[Function]}
                >
                  Search
                </button>
                <ul>
                  <li
                    key="1234"
                    style={
                      Object {
                        "listStyle": "none",
                      }
                    }
                  >
                     
                    <Twit
                      handleDelete={[Function]}
                      twit={
                        Object {
                          "id_str": 1234,
                          "user": Object {},
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "border": "3px solid #b9e0ff",
                            "borderRadius": "10px",
                            "display": "flex",
                            "marginBottom": "30px",
                            "padding": "15px",
                          }
                        }
                      >
                        <div>
                          <img
                            style={
                              Object {
                                "borderRadius": "50%",
                                "height": "auto",
                                "width": "50px",
                              }
                            }
                          />
                        </div>
                        <div
                          style={
                            Object {
                              "marginLeft": " 20px",
                            }
                          }
                        >
                          <span>
                            <strong />
                             @undefined
                          </span>
                          <p
                            style={
                              Object {
                                "margin": "3px 0",
                              }
                            }
                          />
                          <a
                            href="/twit/1234"
                            style={
                              Object {
                                "color": "blue",
                                "fontSize": "16px",
                                "margin": "10px",
                              }
                            }
                          >
                             
                            View
                             
                          </a>
                          <button
                            onClick={[Function]}
                            style={
                              Object {
                                "background": "transparent",
                                "border": "none",
                                "fontSize": "16px",
                                "margin": "10px 5px",
                              }
                            }
                          >
                            Delete
                          </button>
                        </div>
                      </div>
                    </Twit>
                     
                  </li>
                </ul>
              </div>
            </TwitList>
          </Route>
        </Switch>
      </div>
    </App>
  </Router>
</StaticRouter>
`;
